### PR TITLE
LITE-21432:Trim decimal places

### PR DESCRIPTION
**Pull request by KrisMelikova**

### CODE REVIEW

**Review by maxipavlovic — APPROVED** [review]
LGTM, but review from @net-free is required as well.

File: dj_rql/filter_cls.py [hunk 1]
```diff
@@ -890,7 +890,11 @@ def _convert_value(cls, django_field, str_value, use_repr=False):
             return float(val)
 
         elif filter_type == FilterTypes.DECIMAL:
-            return round(float(val), django_field.decimal_places)
+            integer_part = len(val.split('.', 1)[0])
```
Comment (net-free) [inline dj_rql/filter_cls.py:893]: What will be integer_part for negative (-1) value? Should be 1, not 2.

File: dj_rql/filter_cls.py [hunk 2]
```diff
@@ -890,7 +890,11 @@ def _convert_value(cls, django_field, str_value, use_repr=False):
...
+            context = decimal.Context(
+                prec=django_field.decimal_places + integer_part, rounding=decimal.ROUND_DOWN,
```
Comment (net-free) [inline dj_rql/filter_cls.py:895]: Please add additional prec count logic for 0 integer part.
Input 0.123456789 with 4 decimal places filed should give 0.1234, now it's 0.12345.

File: requirements/test.txt [hunk 3]
```diff
@@ -6,6 +6,7 @@ pytest-django==4.4.0
 pytest-mock==3.6.1
 pytest-deadfixtures==2.2.1
 pytest-randomly==3.10.1
+pytz==2021.3
```
Comment (net-free) [inline requirements/test.txt:9]: Is this related to Django 4.0 release deprecated pytz?

**Review by net-free — CHANGES_REQUESTED** [review]

File: dj_rql/filter_cls.py (hunk 2 above)
Comment (KrisMelikova) [reply dj_rql/filter_cls.py]: Thanks, @net-free. Fixed.
My previous solution with "Context" showed unexpected behavior. 'Prec' always skips zero when counting, even if it comes after the dot. Not only when it's a fractional part (for example: 0.012345 -->0.01234, 0.0012345 --> 0.001234)

File: dj_rql/filter_cls.py (hunk 1 above)
Comment (KrisMelikova) [reply dj_rql/filter_cls.py]: Done, thanks.

File: requirements/test.txt (hunk 3 above)
Comment (KrisMelikova) [reply requirements/test.txt:9]: I think so

Comment (net-free) [reply requirements/test.txt:9]: @maxipavlovic i suggest not to change test requirements now and add Django 4 support in separate task.
Do you have any plans for this?

File: dj_rql/filter_cls.py [hunk 4]
```diff
@@ -890,7 +889,10 @@ def _convert_value(cls, django_field, str_value, use_repr=False):
             return float(val)
 
         elif filter_type == FilterTypes.DECIMAL:
-            return round(float(val), django_field.decimal_places)
+            if '.' in val:
+                integer_part, fractional_part = val.split('.', 1)
+                val = integer_part + '.' + fractional_part[:4]
```
Comment (net-free) [inline dj_rql/filter_cls.py:894]:
```python 
fractional_part[:django_field.decimal_places]

**Review by net-free — CHANGES_REQUESTED** [review]

File: dj_rql/filter_cls.py (hunk 4 above)
Comment (KrisMelikova) [reply dj_rql/filter_cls.py]: Done, thanks.

**Review by net-free — APPROVED** [review]

File: requirements/test.txt (hunk 3 above)
Comment (maxipavlovic) [reply requirements/test.txt:9]: I think it's ok to change test requirements in this task, but can be done separately. As you wish :)

**Pull request merged**